Src/SwqlStudio/ConnectionInfo.cs

```diff
@@ -254,28 +255,40 @@ public static T DoWithExceptionTranslation<T>(Func<T> action)
             }
             catch (FaultException ex)
             {
-                msg = (ex.InnerException != null) ? ex.InnerException.Message : ex.Message;
+                msg = ex.InnerException?.Message ?? ex.Message;
                 inner = ex.InnerException ?? ex;
             }
             catch (MessageSecurityException ex)
             {
-                if (ex.InnerException != null && ex.InnerException is FaultException)
+                if (ex.InnerException is FaultException fault)
                 {
-                    msg = (ex.InnerException as FaultException).Message;
-                    inner = ex.InnerException;
+                    msg = fault.Message;
+                    inner = fault;
+                    couldRetry = fault.Code.SubCode.Name == "BadContextToken";
```

Comment:
It looks like subcode might be null -> http://referencesource.microsoft.com/#System.ServiceModel/System/ServiceModel/FaultCode.cs,8597d8aecc900bdb.